proxy/proxy_test.go

```diff
@@ -73,6 +75,87 @@ type listener struct {
 	lastConn chan net.Conn
 }
 
+type blockableBuffer struct {
+	inner *bytes.Buffer
+	m     sync.RWMutex
```

Comment:
Lets use plain mutex here.
Check the docs for RWMutex to better understand the difference to mutex.